database/rrddim.c

```diff
@@ -204,7 +208,6 @@ RRDDIM *rrddim_add_custom(RRDSET *st, const char *id, const char *name, collecte
         return rd;
```

Comment:
You need to handle error cases such as this one. You could add something like:
```
if (host_locked)
    rrdhost_unlock(host);
```

Reply:
I was verifying Netdata code here, the function `rrdhost_wrlock` and `rrdhost_unlock` are definitions to other internal functions, `netdata_rwlock_wrlock_debug` and `netdata_rwlock_unlock_debug`, these other two functions call pthread functions to try lock and try unlock, these functions already have error messages inside it.
I tried to use pthread_rwlock_trywrlock to check errors and I received a lot  `RW_LOCK: failed to obtain write lock (code 35)` and the alarms where not created, os I returned for the current code we have here, because it is working fine.